### PR TITLE
Docs: Contribution guideline for merging pull requests

### PR DESCRIPTION
**Pull request by marefr**
**What this PR does / why we need it**:
Have been thinking for a long time documenting our guidelines for merging pull requests and considerations before merging.  Here is a first draft. Feedback/help appreciated.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: contribute/merge-pull-request.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+# Merge a pull request
```
Comment (kminehart) [inline contribute/merge-pull-request.md:1]: Should we add a section about draft PRs? Is it OK to open a draft PR, should maintainers review drafts, etc

Comment (kminehart) [reply contribute/merge-pull-request.md:1]: right I was looking for a section header somewhere that wasn't about merging that I could add this suggestion to :joy:

File: contribute/merge-pull-request.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+### Assign a milestone
+
+It's recommended to add a milestone to every pull request. This makes it easier to track what changes did go into a certain release. Without this you're basically left with going through git commits which could be a lot harder.
+
+There's also various tooling built that in some cases requires a pull request to be assigned a milestone, for example [generating changelog/release notes](#include-in-changelog-and-release-notes).
+
+### Include in changelog and release notes?
+
+Grafana are generating its [changelog](https://github.com/grafana/grafana/blob/main/CHANGELOG.md) and [release notes](https://grafana.com/docs/grafana/latest/release-notes/) based on merged pull requests. Including changes in the changelog/release notes is very important to provide a somewhat complete picture of what changes a Grafana release actually includes.
```
Comment (hugohaggmark) [inline contribute/merge-pull-request.md:40]:
```suggestion
At Grafana we generate the [changelog](https://github.com/grafana/grafana/blob/main/CHANGELOG.md) and [release notes](https://grafana.com/docs/grafana/latest/release-notes/) based on merged pull requests. Including changes in the changelog/release notes is very important to provide a somewhat complete picture of what changes a Grafana release actually includes.
```

**Review by hugohaggmark — COMMENTED** [review]
Good work on summarizing this. I think it looks great 🎉

**Review by kminehart — APPROVED** [review]

**Pull request merged**